### PR TITLE
Create payout modal

### PR DESCRIPTION
**Pull request by sauravmaiti22**
1. Added modal to create payout.
2. Added validations as per design
3. Added atom for InputTextField to have a numeric variation to  accomdate  account number and sort code which are of a certain max length and are numeric. 
4. Separated out useCreatePayout from useCreateRefund hook as the invalidations are different.
5. Just has one button now to Create the payout, approve is WIP.

### CODE REVIEW

**Comment by sipsorcery** [comment]
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/57.

**Comment by sipsorcery** [comment]
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/57.

File: packages/components/src/components/ui/molecules/Select/SelectBeneficiary/SelectBeneficiary.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
+import { type SelectProps } from '@radix-ui/react-select'
+
+import { LocalBeneficiary } from '../../../../../types/LocalTypes'
+import { cn } from '../../../../../utils'
+// import { cn } from '../../../../../utils'
```
Comment (agranillonf) [inline packages/components/src/components/ui/molecules/Select/SelectBeneficiary/SelectBeneficiary.tsx:5]: We can get rid of this comment.

**Comment by agranillonf** [comment]
There's some validation errors when clicking on "Create" without filling the required fields. It lets you go through and, if the API returns an error, the button remains in the "loading" state, unable to do anything unless the user refreshes the page.
create_payout

**Comment by agranillonf** [comment]
The API requires an account name to be sent. If you don't input anything on it, it'll let you through but the API will return an error stating that. And same as above, the "Create" button will become stuck on the "loading" state.
create_payout_2

**Comment by sauravmaiti22** [comment]
Oops missed it. Thanks @agranillonf.

**Comment by sipsorcery** [comment]
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/57.

**Comment by agranillonf** [comment]
Required field validation has been fixed, but format doesn't seem like it.
create_payout_3

**Comment by sauravmaiti22** [comment]
> Required field validation has been fixed, but format doesn't seem like it. create_payout_3 [ create_payout_3 ](https://user-images.githubusercontent.com/114026805/266718748-a62cdff6-0c89-4b89-904f-711f7d843da8.gif) [ ](https://user-images.githubusercontent.com/114026805/266718748-a62cdff6-0c89-4b89-904f-711f7d843da8.gif)

Yes I noticed that. Should be fixed now.

File: packages/components/src/components/ui/molecules/Select/SelectBeneficiary/SelectBeneficiary.tsx (hunk 1 above)
Comment (sauravmaiti22) [reply packages/components/src/components/ui/molecules/Select/SelectBeneficiary/SelectBeneficiary.tsx]: Done.

**Comment by sipsorcery** [comment]
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/57.

**Review by agranillonf — APPROVED** [review]

**Review by sipsorcery — APPROVED** [review]

File: packages/components/src/components/ui/InputAmountField/InputAmountField.tsx [hunk 2]
```diff
@@ -3,6 +3,7 @@ import * as DropdownMenu from '@radix-ui/react-dropdown-menu'
 import { cva } from 'class-variance-authority'
 import { motion } from 'framer-motion'
 import { InputHTMLAttributes, useEffect, useRef, useState } from 'react'
+// eslint-disable-next-line import/no-unresolved
```
Comment (donalnofrixion) [inline packages/components/src/components/ui/InputAmountField/InputAmountField.tsx:6]: Can you remove this?

**Review by donalnofrixion — APPROVED** [review]

**Comment by donalnofrixion** [comment]
🔥 Looks great Saurav!

One small comment and not super urgent. Can you sort the accounts alphabetically please?
Screenshot 2023-09-09 at 05 13 25

**Review by pabloi-nf — CHANGES_REQUESTED** [review]
The height of all fields have to be the same. Here the dropdown is higher than the text input field. The full height has to be 48px.
Also, the account name font-weight needs to be 400.
image

The dropdown icon is not the correct one.
image
image

The font-weight for all the information inside the beneficiary item has to be 400.
The font-size for the account number and currency has to be 0.75rem.
image

The "add manually" option needs to have a hover state without underline.

The non editable field background color has to be #F6F8F9.

We don't mark the required fields here, only the optional ones.

As far as I understand, the character limit for the reference is 17. Is that correct?

A put random data and I have an error that is not informative. What can be the situation here?
image

Let's remove the back arrow on top. It's better to keep that one for secondary modals, like the refund.

**Comment by sipsorcery** [comment]
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/57.

**Comment by sauravmaiti22** [comment]
> The height of all fields have to be the same. Here the dropdown is higher than the text input field. The full height has to be 48px. Also, the account name font-weight needs to be 400. image
> 
> The dropdown icon is not the correct one. image image
> 
> The font-weight for all the information inside the beneficiary item has to be 400. The font-size for the account number and currency has to be 0.75rem. image
> 
> The "add manually" option needs to have a hover state without underline.
> 
> The non editable field background color has to be #F6F8F9.
> 
> We don't mark the required fields here, only the optional ones.
> 
> As far as I understand, the character limit for the reference is 17. Is that correct?
> 
> A put random data and I have an error that is not informative. What can be the situation here? image
> 
> Let's remove the back arrow on top. It's better to keep that one for secondary modals, like the refund.

Thanks @pabloi-nf. I have fixed all the comments except for "We don't mark the required fields here, only the optional ones.". I couldn't understand this part. About the Create payout failing issue. It was because of invalid IBAN. I added more validation to IBAN now. It should show a message if it's not a valid IBAN.

**Comment by Publicker** [comment]
> Thanks @pabloi-nf. I have fixed all the comments except for "We don't mark the required fields here, only the optional ones.". I couldn't understand this part. About the Create payout failing issue. It was because of invalid IBAN. I added more validation to IBAN now. It should show a message if it's not a valid IBAN.

Hey @sauravmaiti22, I believe @pabloi-nf  was getting at the idea that, unlike the 'Create Payment Request' form where only "Product or service" is marked as required, in this one we highlight the optional fields.

Check out the comparison between the actual implementation and the design:

Here's a comapartion between the implementation and the design:
| Implementation | Design |
|--------|--------|
| Implementation showing required fields | Design - Showing optional fields | 

Hope that clears things up!

**Comment by sauravmaiti22** [comment]
> > Thanks @pabloi-nf. I have fixed all the comments except for "We don't mark the required fields here, only the optional ones.". I couldn't understand this part. About the Create payout failing issue. It was because of invalid IBAN. I added more validation to IBAN now. It should show a message if it's not a valid IBAN.
> 
> Hey @sauravmaiti22, I believe @pabloi-nf was getting at the idea that, unlike the 'Create Payment Request' form where only "Product or service" is marked as required, in this one we highlight the optional fields.
> 
> Check out the comparison between the actual implementation and the design:
> 
> Implementation	Design
> Implementation showing required fields	Design - Showing optional fields
> Hope that clears things up!

@Publicker Thanks Pablo. Yes I discussed it with @pabloi-nf and that was the case. He mentioned that in future we are going to change the implementation in create payment request as well.

**Comment by sipsorcery** [comment]
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/57.

**Comment by Publicker** [comment]
> @Publicker Thanks Pablo. Yes I discussed it with @pabloi-nf and that was the case. He mentioned that in future we are going to change the implementation in create payment request as well.

Brilliant! That makes sense!

Thanks @sauravmaiti.

**Comment by sipsorcery** [comment]
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/57.

**Comment by sipsorcery** [comment]
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/57.

**Review by pabloi-nf — CHANGES_REQUESTED** [review]
I'm seeing a weird animation when 2 warning messages are displayed and one of them dissapears. It seems one replaces the other somehow.

https://github.com/nofrixion/nofrixion.business/assets/127946572/50169edc-7b4f-49e7-81bf-5a632f5f3efb

I can see the "required" text appears before pressing the create button. This message must appear only after submiting, in connection with the "Please fill all required fields" message.
image

**Comment by sipsorcery** [comment]
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/57.

**Review by pabloi-nf — APPROVED** [review]

**Comment by sipsorcery** [comment]
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/57.

**Comment by sipsorcery** [comment]
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/57.

**Pull request merged**